src/main/java/org/mockito/internal/creation/instance/ConstructorInstantiator.java

```diff
@@ -130,4 +127,32 @@ private static boolean paramsMatch(Class<?>[] types, Object[] params) {
         }
         return true;
     }
+
+    private void evaluateConstructor(List<Constructor<?>> matchingConstructors, Constructor<?> constructor) {
```

Comment:
Can you add a javadoc describing this method? It has non trivial complexity and it would be handy to understand why we are implementing the algorithm this way.